x/evm/keeper/state_transition.go

```diff
@@ -351,7 +333,7 @@ func (k *Keeper) ApplyMessage(ctx sdk.Context, msg core.Message, tracer vm.EVMLo
 //
 // If commit is true, the `StateDB` will be committed, otherwise discarded.
 func (k *Keeper) ApplyMessageWithConfig(ctx sdk.Context,
-	msg core.Message,
+	msg *core.Message,
```

Comment:
I don't see the reasoning in the code for using pointers. I think we should avoid using mutable argument

Reply:
The go-ethereum functions accept pointers so I wanted to use the same type. But I think I could switch it to non-pointers too.

Reply:
Note that many of the `core.Message` members are pointers which would still be mutable if removing the top level pointer. We'd need to add a copy/clone function if we wanted to ensure immutability.